src/KOKKOS/mliap_model_python_couple_kokkos.pyx

```diff
@@ -0,0 +1,157 @@
+# cython: language_level=3
+# distutils: language = c++
+
+cimport cython
+
+from libc.stdint cimport uintptr_t
+
+import pickle
+
+# For converting C arrays to numpy arrays
+import numpy
+import torch 
+import cupy
```

Comment:
This unconditionally loads "cupy". I don't see any check for it anywhere.
What will happen if KOKKOS was compiled without CUDA? E.g. configured for HIP or SYCL or OpenMP?

Reply:
Good catch, Will fix.

> E.g. configured for HIP or SYCL or OpenMP
Well, I could say "then you're doing it wrong" :)

Reply:
Added try block for importing cupy.  It will work on GPU for Cuda and do all the CPU/GPU data copies for HIP/SYCL as this is out of my expertise.  To be fair, python is already out of my expertise